src/object.c

```diff
@@ -250,7 +250,7 @@ git_otype git_object_stringn2type(const char *str, size_t len)
 		return GIT_OBJ_BAD;
 
 	for (i = 0; i < ARRAY_SIZE(git_objects_table); i++)
-		if (git_object_typeisloose(i) &&
+		if (*git_objects_table[i].str &&
```

Comment:
Ah, yeah, that line confused me a lot in the last commit. So probably this line change should be moved to the previous commit?

Reply:
Yes, agreed.  I think I fixup'd the wrong commit here.

Reply:
Fixed.